### ISSUE TITLE
Array, Point3d, Vector3d and Transformation need compatibility Overview statements

### ISSUE THREAD

**Issue by DanRathbun** [SketchUp/api-issue-tracker#333]
#### SketchUp Ruby API Documentation Issue

**Array, Point3d, Vector3d and Transformation need compatibility Overview statements**.

----

> _In https://github.com/SketchUp/api-issue-tracker/issues/327#issuecomment-528921091, Dan said:_
>
> Also (documentation wise) the transform argument _(of `Sketchup::Entities#transform_entities`)_ will take a Geom::Vector3d or a 3 element Array object as an implied translational transformation.
>
>Would it be beneficial to document this in such methods that allow this, or a more general note in vector and array class overview paragraphs ?

----
> _In https://github.com/SketchUp/api-issue-tracker/issues/327#issuecomment-529433814, ThomThom said:_
>
> I think this is best to be documented in the Point3d and Vector3d class documentation. Perhaps in some general concept page.
> 
> Adding the array variant overload to all signatures makes the documentation very noisy very quickly. Also makes the signature harder to read:
> ```ruby
> # @param [Geom::Vector3d, Array(Numeric, Numeric, Numeric)] param_name Some possibly verbose description here`
> # @param [Array<Geom::Vector3d, Array(Numeric, Numeric, Numeric)>] param_name Some possibly verbose description here`
>```
> In which case I think it's easier to read the purpose of the param by omitting the implicit array conversions:
> ```ruby
> # @param [Geom::Vector3d] param_name Some possibly verbose description here
> # @param [Array<Geom::Vector3d>] param_name Some possibly verbose description here
>```

Okay, we have this in the Array overview ... which I think is sufficient for Array ...

> The SketchUp Array class adds additional methods to the standard Ruby Array class. Specifically, it contains methods allowing an array to behave just as a `Geom::Vector3d` or `Geom::Point3d` object (which can be thought of as arrays of 3 coordinate values). Therefore, you can use the Array class in place of a `Geom::Point3d` or `Geom::Vector3d` as a way to pass coordinate values.

But it neglects to mention transforms. The overview statement could be appended to say ...

**"Since transformations can use vectors and points for translations, many API methods that have transform argument(s), can accept an array in their place."**

The following statement could also be added to the [`Geom::Transformation`](http://ruby.sketchup.com/Geom/Transformation.html) ...

**"Many API methods that have transformation argument(s), can accept an array in their place for a translational transform, since an array is implicitly convertible to a point or vector."**

----

We have a converse statement in the **[`Geom::Point3d`](http://ruby.sketchup.com/Geom/Point3d.html)** class, but not in the **[`Geom::Vector3d`](http://ruby.sketchup.com/Geom/Vector3d.html)** class. Ie ...
> In addition to the methods below, there are a series of geometry related methods that are on the Array class, since Point3d objects can also be represented as a 3-element Array.

**[`Geom::Vector3d`](http://ruby.sketchup.com/Geom/Vector3d.html)** does not have this statement.

Both [ `Geom`](http://ruby.sketchup.com/Geom.html) classes should have the same statement with perhaps adding (to both) ...

**"Many API methods that have <classname> argument(s), can accept an array in their place."**

~

**Comment by DanRathbun** [comment]
I also have not gotten that deep into the Layout API. It may need the same love for 2D classes and methods.

I notice that [`Geom::Transformation2d`](http://ruby.sketchup.com/Geom/Transformation2d.html) does not even have Overview text.

**Comment by thomthom** [comment]
> I notice that Geom::Transformation2d does not even have Overview text.

 Yes, I really wish the class overviews where better. We've talked about going over these. Chris has a prototype with interactive code examples (if loaded from a web dialog within SketchUp).
